postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresSchemaReader.java

```diff
@@ -58,6 +58,14 @@ public Schema getSchema(ResultSetMetaData metadata, int index) throws SQLExcepti
       return Schema.of(Schema.Type.STRING);
     }
 
+    // If it is a numeric type without precision then use the Schema of String to avoid any precision loss
+    if (Types.NUMERIC == columnType) {
+      int precision = metadata.getPrecision(index);
+      if (precision == 0) {
+        return Schema.of(Schema.Type.STRING);
+      }
+    }
```

Comment:
nit: can we add a log here? similarly as we did for oracle - https://github.com/data-integrations/database-plugins/pull/323/files#diff-dae6a97a5a0b13be65e282c1b7c33fd4e995d3f1a4987ce68e5dbf5d2f1fbc9dR108

Reply:
done